File: frontend/src/app/shared/service/admin/grpe/GroupeEtudiantDetailAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {GroupeEtudiantDetailDto} from '../../../model/grpe/GroupeEtudiantDetail.model';
import {GroupeEtudiantDetailCriteria} from '../../../criteria/grpe/GroupeEtudiantDetailCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";
import {Observable} from "rxjs";


@Injectable({
    providedIn: 'root'
})
export class GroupeEtudiantDetailAdminService extends AbstractService<GroupeEtudiantDetailDto, GroupeEtudiantDetailCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/groupeEtudiantDetail/';
    }

    public findByGroupeEtudiantId(id: number):Observable<Array<GroupeEtudiantDetailDto>>   {
        return this.httpClient.get<Array<GroupeEtudiantDetailDto>>(this.API+'groupeEtudiant/id/'+id)
    }

    public constrcutDto(): GroupeEtudiantDetailDto {
        return new GroupeEtudiantDetailDto();
    }

    public constrcutCriteria(): GroupeEtudiantDetailCriteria {
        return new GroupeEtudiantDetailCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/grpe/GroupeEtudiantDetailAdminService.java
package ma.zs.zyn.service.facade.admin.grpe;

import java.util.List;
import ma.zs.zyn.bean.core.grpe.GroupeEtudiantDetail;
import ma.zs.zyn.dao.criteria.core.grpe.GroupeEtudiantDetailCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface GroupeEtudiantDetailAdminService extends  IService<GroupeEtudiantDetail,GroupeEtudiantDetailCriteria>  {

    List<GroupeEtudiantDetail> findByGroupeEtudiantId(Long id);
    int deleteByGroupeEtudiantId(Long id);
    long countByGroupeEtudiantId(Long id);
    List<GroupeEtudiantDetail> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantRef(String ref);




}
